src/hooks/useSwapTaxes.ts

```diff
@@ -0,0 +1,82 @@
+import { InterfaceEventName } from '@uniswap/analytics-events'
+import { ChainId, Percent } from '@uniswap/sdk-core'
+import { WETH_ADDRESS as getWethAddress } from '@uniswap/universal-router-sdk'
+import { useWeb3React } from '@web3-react/core'
+import FOT_DETECTOR_ABI from 'abis/fee-on-transfer-detector.json'
+import { FeeOnTransferDetector } from 'abis/types'
+import { sendAnalyticsEvent } from 'analytics'
+import { ZERO_PERCENT } from 'constants/misc'
+import { useEffect, useState } from 'react'
+
+import { useContract } from './useContract'
+
+const FEE_ON_TRANSFER_DETECTOR_ADDRESS = '0x19C97dc2a25845C7f9d1d519c8C2d4809c58b43f'
```

Comment:
have you tested this hook while connected to a different chain?  just want to make sure it doesn't cause a crash or something

Reply:
it doesn't crash, but I'll add a check to return early if not on mainnet